webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ComplianceApi.scala

```diff
@@ -98,7 +97,12 @@ class ComplianceApi(
 
       (for {
         level <- restExtractor.extractComplianceLevel(req.params)
-        rules <- complianceService.getRulesCompliance()
+        computeLevel <- Full(if(version.value <= 6) {
+                          None
```

Comment:
Should be Some(10)  (or Some(value expected for api versio 6 )

But Version 6 is not supported anymore with we just may want to remove them (need to check though), but i think 6.1 is 7 minimal api version

Reply:
a few line below you can use compluteLevel insted of level.getOrElse(10)

Reply:
Also this fails if level is not defined, if defined and version > 6 we should default to default vlaue (10)

Reply:
Ok I misread i haven't seen the Full, nevermind this is ok !

Reply:
level is an option that appears only later, so I'd like to keep the semantic of "not defined" , thus an  None

Reply:
Yes i just misread and haven't seen it was in a Full and thought the None could break the computation, but it's not the case at all